kura/org.eclipse.kura.container.orchestration.provider/src/main/java/org/eclipse/kura/container/orchestration/provider/impl/ContainerOrchestrationServiceImpl.java

```diff
@@ -733,4 +717,82 @@ public boolean equals(Object obj) {
+            throw new IllegalArgumentException("Parameters cannot be null or negative");
+        }
+
+        boolean imageAvailableLocally = doesImageExist(imageConfig.getImageName(), imageConfig.getImageTag());
+
+        if (!imageAvailableLocally) {
+            try {
+                imagePullHelper(imageConfig.getImageName(), imageConfig.getImageTag(),
+                        imageConfig.getimageDownloadTimeoutSeconds(), imageConfig.getRegistryCredentials());
+            } catch (InterruptedException e) {
+                throw e;
+            } catch (Exception e) {
+                throw new KuraException(KuraErrorCode.IO_ERROR, "Unable to pull container");
+            }
+        }
+    }
+
+    @Override
+    public void pullImage(String imageName, String imageTag, int timeOutSeconds,
+            Optional<RegistryCredentials> registryCredentials) throws KuraException, InterruptedException {
+        pullImage(new ImageConfiguration.ImageConfigurationBuilder().setImageName(imageName).setImageTag(imageTag)
+                .setImageDownloadTimeoutSeconds(timeOutSeconds).setRegistryCredentials(registryCredentials).build());
+    }
+
+    @Override
+    public List<ImageInstanceDescriptor> listImageInstanceDescriptor() {
+        if (!testConnection()) {
+            throw new IllegalStateException(UNABLE_TO_CONNECT_TO_DOCKER_CLI);
+        }
+        List<Image> images = this.dockerClient.listImagesCmd().withShowAll(true).exec();
+        List<ImageInstanceDescriptor> result = new ArrayList<>();
+        images.forEach(image -> {
+            InspectImageResponse iir = this.dockerClient.inspectImageCmd(image.getId()).exec();
+            result.add(
+                    ImageInstanceDescriptor.builder().setImageName(getImageName(image)).setImageTag(getImageTag(image))
+                            .setImageId(image.getId()).setImageAuthor(iir.getAuthor()).setImageArch(iir.getArch())
+                            .setimageSize(iir.getSize().longValue()).setImageLabels(image.getLabels()).build());
+        });
+        return result;
+    }
+
+    private String getImageName(Image image) {
+        if (image.getRepoTags() == null || image.getRepoTags().length < 1) {
+            return "";
+        }
+
+        return image.getRepoTags()[0].split(":")[0];
+    }
+
+    private String getImageTag(Image image) {
+        if (image.getRepoTags() == null || image.getRepoTags().length < 1) {
+            return "";
+        }
+
+        if (image.getRepoTags()[0].split(":").length < 2) {
+            return "";
+        }
+
+        return image.getRepoTags()[0].split(":")[1];
+    }
+
+    @Override
+    public void deleteImage(String imageId) throws KuraException {
+        checkRequestEnv(imageId);
+        try {
+            this.dockerClient.removeImageCmd(imageId).exec();
+        } catch (Exception e) {
+            logger.error("Could not remove image {}.", imageId);
+            throw new KuraException(KuraErrorCode.OS_COMMAND_ERROR,
```

Comment:
This error requires two parameters: https://github.com/eclipse/kura/blob/d8ce302a04c930dc534c7d4f1af68ada82e8d43b/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraErrorCode.java#L148
So be aware that the error managing will format the message as described here: https://github.com/eclipse/kura/blob/d8ce302a04c930dc534c7d4f1af68ada82e8d43b/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraErrorCode.java#L144.